pkg/controller/logstorage/linseed/linseed_controller.go

```diff
@@ -0,0 +1,413 @@
+		return reconcile.Result{}, err
+	}
+
+	var esClusterConfig *relasticsearch.ClusterConfig
+	if managementClusterConnection == nil {
+		flowShards := logstoragecommon.CalculateFlowShards(args.LogStorage.Spec.Nodes, logstoragecommon.DefaultElasticsearchShards)
+		esClusterConfig = relasticsearch.NewClusterConfig(render.DefaultElasticsearchClusterName, args.LogStorage.Replicas(), logstoragecommon.DefaultElasticsearchShards, flowShards)
+	}
+
+	// Query the username and password this Linseed instance should use to authenticate with Elasticsearch.
+	// These credentials are created by the elasticsearch users controller. Delay installing Linseed until available.
+	key := types.NamespacedName{Name: render.ElasticsearchLinseedUserSecret, Namespace: request.TruthNamespace()}
+	if err = r.client.Get(ctx, key, &corev1.Secret{}); err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, fmt.Sprintf("Error getting Secret %s", key), err, reqLogger)
+		return reconcile.Result{}, err
+	} else if errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceNotFound, fmt.Sprintf("Waiting for Linseed credential Secret %s", key), err, reqLogger)
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Determine the namespaces to which we must bind the linseed cluster role.
+	bindNamespaces := []string{request.InstallNamespace()}
+	if r.multiTenant {
+		bindNamespaces, err = utils.TenantNamespaces(ctx, r.client)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	cfg := &linseed.Config{
+		Installation:      install,
+		PullSecrets:       pullSecrets,
+		Namespace:         request.InstallNamespace(),
+		BindNamespaces:    bindNamespaces,
+		TrustedBundle:     trustedBundle,
+		ClusterDomain:     r.clusterDomain,
+		KeyPair:           linseedKeyPair,
+		TokenKeyPair:      tokenKeyPair,
+		UsePSP:            r.usePSP,
+		ESClusterConfig:   esClusterConfig,
+		ManagementCluster: managementCluster != nil,
+		Tenant:            args.Tenant,
+	}
+	linseedComponent := linseed.Linseed(cfg)
+
+	if err := imageset.ApplyImageSet(ctx, r.client, variant, linseedComponent); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// In standard installs, the LogStorage owns Linseed. For multi-tenant, it's owned by the Manager instance.
+	var hdler utils.ComponentHandler
+	if r.multiTenant {
+		hdler = utils.NewComponentHandler(reqLogger, r.client, r.scheme, args.Tenant)
+	} else {
+		hdler = utils.NewComponentHandler(reqLogger, r.client, r.scheme, args.LogStorage)
+	}
+	if err := hdler.CreateOrUpdateOrDelete(ctx, linseedComponent, r.status); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating / deleting resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if !r.multiTenant {
```

Comment:
These should probably all just be in another controller. Clean separation of controllers that need to handle multi-tenancy, and those that don't.

Reply:
Assuming it ends up that we don't need these in multi-tenant mgmt clusters.